sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.netstandard2.0.cs

```diff
@@ -494,24 +531,32 @@ public partial class ShareMetrics
+        public static Azure.Storage.Files.Shares.Models.FileBreakLeaseResult FileBreakLeaseResult(Azure.ETag eTag, System.DateTimeOffset lastModified, string leaseId, string clientRequestId) { throw null; }
+        public static Azure.Storage.Files.Shares.Models.FileChangeLeaseResult FileChangeLeaseResult(Azure.ETag eTag, System.DateTimeOffset lastModified, string leaseId, string clientRequestId) { throw null; }
+        public static Azure.Storage.Files.Shares.Models.FileReleaseLeaseResult FileReleaseLeaseResult(Azure.ETag eTag, System.DateTimeOffset lastModified, string clientRequestId) { throw null; }
         public static Azure.Storage.Files.Shares.Models.PermissionInfo PermissionInfo(string filePermissionKey) { throw null; }
         public static Azure.Storage.Files.Shares.Models.ShareFileCopyInfo ShareFileCopyInfo(Azure.ETag eTag, System.DateTimeOffset lastModified, string copyId, Azure.Storage.Files.Shares.Models.CopyStatus copyStatus) { throw null; }
         public static Azure.Storage.Files.Shares.Models.ShareFileHandle ShareFileHandle(string handleId, string path, string fileId, string sessionId, string clientIp, string parentId = null, System.DateTimeOffset? openedOn = default(System.DateTimeOffset?), System.DateTimeOffset? lastReconnectedOn = default(System.DateTimeOffset?)) { throw null; }
         public static Azure.Storage.Files.Shares.Models.ShareFileRangeInfo ShareFileRangeInfo(System.DateTimeOffset lastModified, Azure.ETag eTag, long fileContentLength, System.Collections.Generic.IEnumerable<Azure.HttpRange> ranges) { throw null; }
         public static Azure.Storage.Files.Shares.Models.ShareFileUploadInfo ShareFileUploadInfo(Azure.ETag eTag, System.DateTimeOffset lastModified, byte[] contentHash, bool isServerEncrypted) { throw null; }
         public static Azure.Storage.Files.Shares.Models.ShareInfo ShareInfo(Azure.ETag eTag, System.DateTimeOffset lastModified) { throw null; }
         public static Azure.Storage.Files.Shares.Models.ShareItem ShareItem(string name, Azure.Storage.Files.Shares.Models.ShareProperties properties, string snapshot = null) { throw null; }
-        public static Azure.Storage.Files.Shares.Models.ShareProperties ShareProperties(System.DateTimeOffset? lastModified = default(System.DateTimeOffset?), Azure.ETag? eTag = default(Azure.ETag?), int? quotaInGB = default(int?), System.Collections.Generic.IDictionary<string, string> metadata = null) { throw null; }
+        public static Azure.Storage.Files.Shares.Models.ShareProperties ShareProperties(System.DateTimeOffset? lastModified = default(System.DateTimeOffset?), Azure.ETag? eTag = default(Azure.ETag?), int? provisionedIops = default(int?), int? provisionedIngressMBps = default(int?), int? provisionedEgressMBps = default(int?), System.DateTimeOffset? nextAllowedQuotaDowngradeTime = default(System.DateTimeOffset?), int? quotaInGB = default(int?), System.Collections.Generic.IDictionary<string, string> metadata = null) { throw null; }
```

Comment:
This is a breaking change as the parameters are re-ordered.

Reply:
This was auto-generated, I'm not sure there's a way around the breaking change since we have to add more parameters.

Reply:
Some of that class is generated and some of it's written by hand (it's a partial class).  In this case, to avoid the break you should copy the old signature into a partial class in the .\Models directory and forward it on to the new generated method with appropriate defaults.  This is the sort of thing that will need to be supported by a multi-version code generator at some point in the future, but for now we'll have to maintain compatibility by hand.

Reply:
Fixed.